src/centralstore/incremental/PropertyLink.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+Copyright 2020 JasmineGraph Team
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+**/
+
+#include "PropertyLink.h"
+
+#include <sstream>
+#include <vector>
+
+#include "../../util/logger/Logger.h"
+
+Logger property_link_logger;
+
+PropertyLink::PropertyLink(unsigned int propertyBlockAddress) : blockAddress(propertyBlockAddress) {
+    if (propertyBlockAddress > 0) {
+        this->propertiesDB->seekg(propertyBlockAddress);
+        char rawName[PropertyLink::MAX_NAME_SIZE] = {0};
+        // char rawValue[PropertyLink::MAX_VALUE_SIZE] = {0};
+
+        if (!this->propertiesDB->read(reinterpret_cast<char*>(&rawName), PropertyLink::MAX_NAME_SIZE)) {
+            property_link_logger.error("Error while reading property name from block " + std::to_string(blockAddress));
+        }
+        property_link_logger.debug("Current FD curser position = " + std::to_string(this->propertiesDB->tellg()) +
+                                   " when reading = " + std::to_string(blockAddress));
+        if (!this->propertiesDB->read(reinterpret_cast<char*>(&this->value), PropertyLink::MAX_VALUE_SIZE)) {
+            property_link_logger.error("Error while reading property value from block " + std::to_string(blockAddress));
+        }
+
+        if (!this->propertiesDB->read(reinterpret_cast<char*>(&(this->nextPropAddress)), sizeof(unsigned int))) {
+            property_link_logger.error("Error while reading property next address from block " +
+                                       std::to_string(blockAddress));
+        }
+
+        this->name = std::string(rawName);
+        // this->value = rawValue;
+    }
+};
+
+PropertyLink::PropertyLink(unsigned int blockAddress, std::string name, char* rvalue, unsigned int nextAddress)
+    : blockAddress(blockAddress), nextPropAddress(nextAddress), name(name) {
+    // Can't use just string copyer here because of binary data formats
+    for (size_t i = 0; i < PropertyLink::MAX_VALUE_SIZE; i++) {
+        this->value[i] = rvalue[i];
+    }
+};
+
+/**
+ * pseudo code for inserting an element to a property (single) link list
+ * ### Note: This link list implimentation adds a new link to the tail of the link, unlike in normal case relocating the
```

Comment:
Change `implimentation` to `implementation`

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/ed31ba0af33e9629173808960eef3cfe90c5dbe9

Reply:
Done